src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -255,15 +256,34 @@ private async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdenti
                 tasks.Add(_localRepository.GetPackageMetadataFromLocalSourceAsync(identity, cancellationToken));
             }
 
+            if (_globalLocalRepositories != null)
+            {
+                _globalLocalRepositories.ForEach(x =>
+                    tasks.Add(x.GetPackageMetadataFromLocalSourceAsync(identity, cancellationToken)));
+            }
+
             IEnumerable<IPackageSearchMetadata> completed = (await Task.WhenAll(tasks))
                 .Where(m => m != null);
 
-            IPackageSearchMetadata master = completed.FirstOrDefault(m => !string.IsNullOrEmpty(m.Summary))
-                ?? completed.FirstOrDefault()
-                ?? PackageSearchMetadataBuilder.FromIdentity(identity).Build();
+            PackageSearchMetadataBuilder metadataBuilder = completed.Where(m => !string.IsNullOrEmpty(m.Summary)).Select(PackageSearchMetadataBuilder.FromMetadata).FirstOrDefault()
+                ?? completed.Select(PackageSearchMetadataBuilder.FromMetadata).FirstOrDefault()
+                ?? PackageSearchMetadataBuilder.FromIdentity(identity);
 
-            return master.WithVersions(
-                asyncValueFactory: () => MergeVersionsAsync(identity, completed));
+            var clonedResult = metadataBuilder
+                  .WithVersions(AsyncLazy.New(() => MergeVersionsAsync(identity, completed)))
+                  .Build() as PackageSearchMetadataBuilder.ClonedPackageSearchMetadata;
+
+            if (string.IsNullOrWhiteSpace(clonedResult.ReadmeFileUrl))
+            {
+                clonedResult.ReadmeFileUrl = completed.Select(m => m.ReadmeFileUrl).FirstOrDefault(u => !string.IsNullOrWhiteSpace(u));
+            }
+
+            if (string.IsNullOrWhiteSpace(clonedResult.PackagePath))
+            {
+                clonedResult.PackagePath = completed.Select(m => (m as PackageSearchMetadataBuilder.ClonedPackageSearchMetadata)?.PackagePath).FirstOrDefault(u => !string.IsNullOrWhiteSpace(u));
```

Comment:
Even though it's just a LINQ expression, it's good to use clear variable names. Is the `u` supposed to indicate a URI string?

```suggestion
                clonedResult.PackagePath = completed.Select(m => (m as PackageSearchMetadataBuilder.ClonedPackageSearchMetadata)?.PackagePath).FirstOrDefault(uri => !string.IsNullOrWhiteSpace(uri));
```